pkg/machines/components/vmSnapshotsTab.jsx

```diff
@@ -43,118 +44,153 @@ function prettyTime(unixTime) {
+        this.setState({ showCreateSnapshotModal: false });
+    }
+
+    render() {
+        const { vm, dispatch } = this.props;
+        const id = vmId(vm.name);
+
+        const emptyCaption = (<div className="no-snapshots">
+            <h3>{_("No snapshots")}</h3>
+            {_("Previously taken snapshots allow you to revert to an earlier state if something goes wrong")}
+        </div>);
+
+        let detailMap = [
+            {
+                name: _("Creation Time"), value: (snap, snapId) => {
+                    const date = prettyTime(snap.creationTime);
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-date`}>
+                            {date}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("Name"), value: (snap, snapId) => {
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-name`}>
+                            {snap.name}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("Description"), value: (snap, snapId) => {
+                    let desc = snap.description;
+                    if (!desc)
+                        desc = (<span className="snap-greyed-out">{_("No description")}</span>);
+
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-description`}>
+                            {desc}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("State"), value: (snap, snapId) => {
+                    const statesMap = {
+                        shutoff: "shut off",
+                        "disk-snapshot": "disk",
+                    };
+                    const state = statesMap[snap.state] || snap.state;
+
+                    const infoTips = {
+                        shutdown: _("Shutting down"),
+                        disk: _("Disk-only snapshot"),
+                        blocked: _("Domain is blocked on resource"),
+                        crashed: _("Domain has crashed"),
+                    };
+                    const tooltipMessage = infoTips[state];
+                    const tooltip = tooltipMessage
+                        ? (<span className="tooltip-circle">
+                            <Tooltip entryDelay={0} exitDelay={0} content={tooltipMessage}>
+                                <InfoAltIcon />
+                            </Tooltip>
+                        </span>) : null;
+
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-type`}>
+                            {state}
+                            {tooltip}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("Parent snapshot"), value: (snap, snapId) => {
```

Comment:
These  capitalization changes should be in different commit.

Reply:
Fixed this and all above